newIDE/app/src/EventsSheet/ParameterFields/ObjectVariableField.js

```diff
@@ -45,17 +45,49 @@ export default React.forwardRef<ParameterFieldProps, ParameterFieldInterface>(
     });
 
     const { layout } = scope;
-    let object = null;
-    let variablesContainer = null;
-    if (objectName) {
-      if (layout && layout.hasObjectNamed(objectName)) {
-        object = layout.getObject(objectName);
-        variablesContainer = object.getVariables();
-      } else if (project && project.hasObjectNamed(objectName)) {
-        object = project.getObject(objectName);
-        variablesContainer = object.getVariables();
-      }
-    }
+    const object = objectName
+      ? layout && layout.hasObjectNamed(objectName)
+        ? layout.getObject(objectName)
+        : project && project.hasObjectNamed(objectName)
+        ? project.getObject(objectName)
+        : null
+      : null;
+    const variablesContainers = React.useMemo<Array<gdVariablesContainer>>(
+      () => {
+        if (!objectName) {
+          return [];
+        }
+        const variablesContainers: Array<gdVariablesContainer> = [];
+        if (object) {
+          variablesContainers.push(object.getVariables());
+        } else {
+          const group =
+            layout && layout.getObjectGroups().has(objectName)
+              ? layout.getObjectGroups().get(objectName)
+              : project && project.getObjectGroups().has(objectName)
+              ? project.getObjectGroups().get(objectName)
+              : null;
+
+          if (group) {
+            for (const subObjectName of group
+              .getAllObjectsNames()
+              .toJSArray()) {
+              if (layout && layout.hasObjectNamed(subObjectName)) {
+                variablesContainers.push(
+                  layout.getObject(subObjectName).getVariables()
+                );
+              } else if (project && project.hasObjectNamed(subObjectName)) {
+                variablesContainers.push(
+                  project.getObject(subObjectName).getVariables()
+                );
+              }
+            }
+          }
+        }
```

Comment:
You can add a comment telling that this should probably be moved to a function `GetObjectOrGroupVariablesContainers` inside `ObjectsContainersList` (even if we don't do it now)